src/store/index.ts

```diff
@@ -160,6 +160,12 @@ const {
 
     setSelected(state, selectedNodes: Set<string>) {
       state.selectedNodes = selectedNodes;
+
+      if (state.provenance !== null) {
+        if (selectedNodes.size === 0) {
+          updateProvenanceState(state, 'Clear Selection');
+        }
```

Comment:
It seems like this condition will cause "clear selection" to appear in the provenance history even when simply de-selecting a single node. That's ok, but it's not precisely what the "clear selection" action means, so I'm mentioning here in case you didn't think of it.

Reply:
Adding and removing one node use a different function here, namely `addSelectedNode` and `removeSelectedNode` so the problem you mentioned shouldn't ever happen.

There could be another issue with the provenance here though, since this is used to reset the state when the provenance changes. I'll check and report back

Reply:
And the provenance should be fine, it doesn't use these functions. It accesses the state directly with `provState.selectedNodes = [...newProvState.selectedNodes] as any;` in provenanceUtils.ts